lib/rgl/dot.rb

```diff
@@ -52,55 +50,42 @@ def to_dot_graph(params = {})
       fontsize       = params['fontsize'] ? params['fontsize'] : '8'
       graph          = (directed? ? DOT::Digraph : DOT::Graph).new(params)
       edge_class     = directed? ? DOT::DirectedEdge : DOT::Edge
-      vertex_options = params['vertex'] || {}
-      edge_options   = params['edge'] || {}
 
       each_vertex do |v|
         default_vertex_options = {
           'name'     => vertex_id(v),
           'fontsize' => fontsize,
           'label'    => vertex_label(v)
         }
-        each_vertex_options = default_vertex_options.merge(vertex_options)
-
-
-
-        vertex_options.each do |option, val|
-          each_vertex_options[option] = if val.is_a?(Proc)
-                                          if val.call(v).nil?
-                                            nil
-                                          elsif val.call(v).key?(:"#{option}")
-                                            val.call(v)[:"#{option}"]
-                                          end
-                                        else
-                                          val
-                                        end
-        end
-
-          graph << DOT::Node.new(each_vertex_options)
+        each_vertex_options = default_vertex_options
+
+        if @vertex_options && @vertex_options[v]
+          RGL::DOT::NODE_OPTS.each do |opt|
+            if @vertex_options[v].key?(:"#{opt}")
```

Comment:
Could we perhaps use symbols instead of strings in the OPTS constants? That would spare the conversion here. When the user sets options we could allow both string and symbols. Strings would be converted when setting the value.

I am not sure...

Reply:
I can't find the mentioned original `dot.rb` file from the rdoc project that the modified version was derived from so I couldn't see if it had a purpose looking like this. I think symbols should be fine since it's just a big array of options that aren't really used anywhere else from what I can tell.

Could you give me an example of how you would imagine a user to pass a string to the method?

Something like?
```
graph.set_vertex_options('a', label = "My Label")
```

Reply:
I looked briefly into converting the OPTS constants to symbols and it will break all the rdot tests and certainly need a lot more work to figure out properly. Specifically the "pass me a string that contains both the keyword/symbol and value" to take apart could be quite expensive (in terms of edge cases) when you think about labels or other desired node/edge content strings containing `=`. Maybe this proves trivial to someone else, but I would shy away from this right now. The methods are well defined and allow simple usage for setting complex values without much interpretation. To me, that's good enough.

Reply:
Yes, but symbols are fine. I am not sure, if strings were allowed in the previous version. If so, we should allow them. If not, we can stick with symbols at API level.

Internally having symbols in the OPTS arrays would be an optimization. But since we are dealing only with small graphs, this also is not necessary. I leave it up to you, to dig further.

Reply:
My suggestion would be to merge the current state and I'll look into it when I have more time for it. This could be a separate improvement.